packages/node-http-handler/src/node-http2-handler.ts

```diff
@@ -118,15 +141,45 @@ export class NodeHttp2Handler implements HttpHandler {
     });
   }
 
+  /**
+   * Returns a new session for the given URL.
+   * @param authority The URL to create a session for.
+   * @returns A new session for the given URL.
+   */
   private getSession(authority: string): ClientHttp2Session {
-    const connectionPool = this.connectionPool;
+    const newSession = connect(authority);
+    const destroySessionCb = () => {
+      this.destroySession(newSession);
+    };
+    newSession.on("goaway", destroySessionCb);
+    newSession.on("error", destroySessionCb);
+    newSession.on("frameError", destroySessionCb);
+
+    const sessionTimeout = this.sessionTimeout;
+    if (sessionTimeout) {
+      newSession.setTimeout(sessionTimeout, () => {
+        newSession.close();
```

Comment:
In case of connection timeout, we should also remove it from the session cache.

Reply:
It's removed on Line 182 in `getSessionFromCache`

Reply:
We should remove it from `sessionList` here.